warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -1337,6 +1339,14 @@ protected IteratorBuildingVisitor createIteratorBuildingVisitor(Class<? extends
         // TODO: .setStatsPort(this.statsdHostAndPort);
     }
     
+    protected ArrayBlockingQueue<SortedKeyValueIterator<Key,Value>> createIvaratorSourcePool(SortedKeyValueIterator<Key,Value> sourceForDeepCopies,
+                    int maxIvaratorSources, IteratorEnvironment env) {
+        ArrayBlockingQueue<SortedKeyValueIterator<Key,Value>> ivaratorSourcePool = new ArrayBlockingQueue<>(maxIvaratorSources, true);
+        for (int i = 0; i < maxIvaratorSources; i++)
+            ivaratorSourcePool.add(sourceForDeepCopies.deepCopy(env));
```

Comment:
does this deepCopy do the opening of the RFiles, or is that lazy and not done until the first seek?  Just curious if we need to create these lazily.

Reply:
Now you've done it...  Fix coming soon...